src/main/java/com/scalar/db/storage/dynamo/Dynamo.java

```diff
@@ -54,7 +54,8 @@ public Dynamo(DatabaseConfig config) {
         DynamoDbClient.builder()
             .credentialsProvider(
                 StaticCredentialsProvider.create(
-                    AwsBasicCredentials.create(config.getUsername(), config.getPassword())))
+                    AwsBasicCredentials.create(
+                        config.getUsername().orElse(null), config.getPassword().orElse(null))))
```

Comment:
If both are null, does it assume that the code uses credentials stored in a file (like ~/aws/config) implicitly?
I'm just wondering if there is a case where it works without a username and a password.

Reply:
If any of them is null, the DynamoDbClient builder will throw NullPointerException with a message. I thought it was okay as the same reason above. BTW, the current implementation doesn't check username and password, so if username or password is empty, we will also face NullPointerException.